UM-USElab/gradecraft-development
# Import Canvas Users

### Status
READY

### Description
We currently can import various things such as grades and assignments from Canvas, but we cannot import only the users for a course. This PR adds this ability.

The user import process is structured in a way such that it can import all users for a particular course and not just the students. Since Canvas allows for the possibility of having multiple roles for a course, for our purposes only the most "important" role will be created. In order of precedence, the Canvas roles are as follows:

`TeacherEnrollment > DesignerEnrollment > TaEnrollment > StudentEnrollment > ObserverEnrollment`

Note that at this time, the designer role is mapped to a GSI role in Gradecraft.

### Todos
- [x] Add Documentation (Documented in #3263)

### Gem dependencies
N/A

### Migrations
NO

### Steps to Test or Reproduce
Head to the Users page where you will now notice the option to Import Users in the upper right-hand corner. Click and select to import users from Canvas.

Ensure that all users load via pagination and that upon submitting the form, the users are created successfully and their course roles properly assigned.

### Impacted Areas in Application
Also impacts Canvas grade import, since refactoring was required in order to reuse some existing user import logic.

======================
Closes #2934
